homeassistant/components/template/alarm_control_panel.py

```diff
@@ -254,13 +254,14 @@ def _update_state(self, result):
         )
         self._state = None
 
-    async def async_added_to_hass(self) -> None:
-        """Register callbacks."""
+    @callback
+    def _async_setup_templates(self) -> None:
+        """Set up templates."""
         if self._template:
             self.add_template_attribute(
                 "_state", self._template, None, self._update_state
             )
-        await super().async_added_to_hass()
+        super()._async_setup_templates()
```

Comment:
Shouldn't we call super first and then the code in the override? If the override overrides some attribute settings the override should normally be run after the default.

Reply:
Yes we should, but since this is an old issue I'd prefer to fix it in a separate PR.